### PR TITLE
hw8 jdi

### PR DESCRIPTION
**Pull request by ErmolinFedor**
jdi

### CODE REVIEW

File: src/main/java/hw8/beans/Data.java [hunk 1]
```diff
@@ -0,0 +1,123 @@
+
+package hw8.beans;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class Data {
```
Comment (troe88) [inline src/main/java/hw8/beans/Data.java]: What kind of data is this ?

File: src/main/java/hw8/beans/Data.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        return vegetables;
+    }
+
+    public void setVegetables(List<String> vegetables) {
+        this.vegetables = vegetables;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
```
Comment (troe88) [inline src/main/java/hw8/beans/Data.java]: Oh my, are you serious ? I mentioned twice, at least, the fact that you should check common methods out before you are going to implement your own one. Take a look at the Strings::join

File: src/main/java/hw8/beans/Data.java [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+        if (sb.charAt((sb.length()- 1)) == ',') {
+            sb.setCharAt((sb.length()- 1), ']');
+        } else {
+            sb.append(']');
+        }
+        return sb.toString();
+    }
+
+    @Override
+    public int hashCode() {
```
Comment (troe88) [inline src/main/java/hw8/beans/Data.java]: Is this auto generated method ?

File: src/main/java/hw8/beans/DataSet.java [hunk 4]
```diff
@@ -0,0 +1,100 @@
+
+package hw8.beans;
+
+import com.google.gson.annotations.Expose;
+import com.google.gson.annotations.SerializedName;
+
+public class DataSet {
```
Comment (troe88) [inline src/main/java/hw8/beans/DataSet.java]: This is completely prohibited to work with data in this way. Imagine the situation if the data set will be changed.

Comment (troe88) [reply src/main/java/hw8/beans/DataSet.java]: Besides, this class is useless.

File: src/test/java/hw8/JdiTests.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    public void beforeSuite() {
+        initElements(JdiSite.class);
+    }
+
+    @AfterSuite(alwaysRun = true)
+    public void afterSuite() {
+        killAllSeleniumDrivers();
+    }
+
+    @Test(priority =0)
```
Comment (troe88) [inline src/test/java/hw8/JdiTests.java]: What is the reason of two tests here ?

File: src/test/java/hw8/JdiTests.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+        JdiSite.jdiHomePage.login(ROMAN);
+        JdiSite.jdiHomePage.userName.is().text(ROMAN.getName());
+    }
+
+
+    @Test(priority = 1, dataProvider = "MetalsColorsToDo", dataProviderClass =MetalsColorsData.class)
+    public void jdiMetalsColorsTest(Data data, List<String> expectedList) {
+        JdiSite.jdiHomePage.clickMetalsColors();
+        JdiSite.jdiMetalsColorsPage.checkOpened();
+        JdiSite.jdiMetalsColorsPage.fillMetalsColorsForm(data);
```
Comment (troe88) [inline src/test/java/hw8/JdiTests.java]: Why don't you create one single method Form::submit(entity) instead of two method fill(entity) and  submit()

**Review by troe88 — CHANGES_REQUESTED** [review]

File: src/main/java/hw8/beans/Data.java (hunk 1 above)
Comment (ErmolinFedor) [reply src/main/java/hw8/beans/Data.java]: renamed to MetalsColors

File: src/main/java/hw8/beans/DataSet.java (hunk 4 above)
Comment (ErmolinFedor) [reply src/main/java/hw8/beans/DataSet.java]: removed

File: src/test/java/hw8/JdiTests.java (hunk 5 above)
Comment (ErmolinFedor) [reply src/test/java/hw8/JdiTests.java]: Deleted the extra ones.

File: src/test/java/hw8/JdiTests.java (hunk 6 above)
Comment (ErmolinFedor) [reply src/test/java/hw8/JdiTests.java]: I agree.

**Review by troe88 — APPROVED** [review]

**Pull request merged**